File: Examples/test-suite/perl5/extend_typedef_class_runme.pl
use strict;
use warnings;
use Test::More tests => 10;
BEGIN { use_ok('extend_typedef_class') }
require_ok('extend_typedef_class');

# No namespace
{
  my $s = extend_typedef_class::AClass->new();
  $s->{membervar} = 10;
  is($s->getvar(), 10, 'AClass');
}
{
  my $s = extend_typedef_class::BClass->new();
  $s->{membervar} = 20;
  is($s->getvar(), 20, 'BClass');
}
{
  my $s = extend_typedef_class::CClass->new();
  $s->{membervar} = 30;
  is($s->getvar(), 30, 'CClass');
}
{
  my $s = extend_typedef_class::DClass->new();
  $s->{membervar} = 40;
  is($s->getvar(), 40, 'DClass');
}
# In namespace
{
  my $s = extend_typedef_class::AStruct->new();
  $s->{membervar} = 10;
  is($s->getvar(), 10, 'AStruct');
}
{
  my $s = extend_typedef_class::BStruct->new();
  $s->{membervar} = 20;
  is($s->getvar(), 20, 'BStruct');
}
{
  my $s = extend_typedef_class::CStruct->new();
  $s->{membervar} = 30;
  is($s->getvar(), 30, 'CStruct');
}
{
  my $s = extend_typedef_class::DStruct->new();
  $s->{membervar} = 40;
  is($s->getvar(), 40, 'DStruct');
}

File: Examples/test-suite/lua/extend_typedef_class_runme.lua
require("import")
require("extend_typedef_class")
etc=extend_typedef_class
catch_undef_globs() -- catch "undefined" global variables

function test_obj(obj, const)
  obj.membervar = const
  assert(obj:getvar() == const)
end

a1 = etc.AClass()
test_obj(a1,1)

b1 = etc.BClass()
test_obj(b1,2)

c1 = etc.CClass()
test_obj(c1,3)

d1 = etc.DClass()
test_obj(d1,4)

a2 = etc.AStruct()
test_obj(a2,5)

b2 = etc.BStruct()
test_obj(b2,6)

c2 = etc.CStruct()
test_obj(c2,7)

d2 = etc.DStruct()
test_obj(d2,8)
